core/src/wallet/cosmos/explorers/GaiaCosmosLikeBlockchainExplorer.cpp

```diff
@@ -1218,13 +1218,25 @@ FuturePtr<BigInt> GaiaCosmosLikeBlockchainExplorer::getEstimatedGasLimit(
     /// directly the gas needed for a transaction with several messages.
     /// As a workaround, we split the transaction in several transactions
     /// containing only one message and sum the costs of all transactions.
-    /// This implies that the cost of the signature is paid as many times
-    /// as there are messages, instead of one. We assume that this cost is
-    /// low compared to the total cost.
+    ///
+    /// This creates various approximations in the total gas cost, but we still found
+    /// out that a simple MsgSend transaction can run out of this gas estimation
+    /// even with a gasAdjustment of 2.
+    ///
+    /// Therefore we probably need to adjust the gas assuming that we are
+    /// underestimating the cost.
+    ///
+    /// The starting value of the accumulate is the cost of verifying a
+    /// SECP256k1 signature, and the cost of the size of the transaction.
+    const uint32_t SigVerifyCostSecp256k1 = 1000;
+    const uint32_t SigSize = 64;  // A signature is approx 64 bytes long
+    const uint32_t TxSizeCost = 10 * (transaction->serializeForSignature().size() + SigSize);
+    const BigInt BaseTxGasCost(SigVerifyCostSecp256k1 + TxSizeCost);
+
     return async::sequence(getContext(), estimations)
-        .flatMapPtr<BigInt>(getContext(), [](const auto &estimations) {
+        .flatMapPtr<BigInt>(getContext(), [BaseTxGasCost](const auto &estimations) {
```

Comment:
Since BaseTxGasCost is a static constant, you shouldn't need to capture it.

Reply:
It depends on `transaction->serializeForSignature.size()` so I need the capture here (either capturing BaseTxGasCost, or capturing `transaction` to do this computation in the closure)

Reply:
I agree with @gagbo.
`BaseTxGasCost` isn't a static constant variable here and the compiler should complain if it was the case.